client/tests/simulator_tests.rs

```diff
@@ -83,6 +83,63 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
```

Comment:
Test structure idea: I can see this is going to get unweildy as more commands are implemented. WDYT of each command having a separate module and then you can implement tests specific to commands in those submodules?

Then you can name the tests more specifically what they test and it'll look nicer in test output.

So for instance, `simulator_tests::get_random::trivial_repeat_check` would be this test, and `simulator_tests::get_random::clamps_to_tpm2_digest_size` would be the other test I've asked for below.

Reply:
I think it is a great idea. We already moved all command structures into sub folders. And these tests may explode in future. I will move it once we are done reviewing the current test cases 👍 🙂 Or we can create a separate task for this cleanup, no to have too many changes in one go.

Reply:
I left a separate comment here around the oddities of `tests` module resolution. I kind of forgot about that, so I hope it didn't cause too much headaches... Anyways, I think it's worth the headaches, less we eventually have a several thousand-line `simulator_tests.rs` file which will be impossible to maintain...

Thanks for restructuring!